### PR TITLE
OTT-1146: Add prometheus stats for vast unwrap module

### PR DESCRIPTION
**Pull request by Pubmatic-Supriya-Patil**
# Description

Please add change description or link to ticket, docs, etc.

# Checklist:

- [ ] PR commit list is unique (rebase/pull with the origin branch to keep master clean).
- [ ] JIRA number is added in the PR title and the commit message.
- [ ] Updated the `header-bidding` repo with appropiate commit id.
- [ ] Documented the new changes.

For Prebid upgrade, refer: https://inside.pubmatic.com:8443/confluence/display/Products/Prebid-server+upgrade

### CODE REVIEW

File: modules/pubmatic/vastunwrap/unwrap_service.go [hunk 1]
```diff
@@ -12,33 +12,46 @@ import (
...
 	wrap_cnt := httpResp.Header().Get(UnwrapCount)
+	respStatus := httpResp.Header().Get("unwrap-status")
 	if wrap_cnt != "" {
 		wrapperCnt, _ = strconv.Atoi(wrap_cnt)
 	}
 	respBody := httpResp.Body.Bytes()
 	if httpResp.Code == http.StatusOK {
 		bid.Bid.AdM = string(respBody)
 		glog.Infof("\n UnWrap Done for BidId = %s Cnt = %d in %v (ms)", bid.Bid.ID, wrapperCnt, time.Since(startTime).Milliseconds())
+		m.MetricsEngine.RecordRequestStatus(accountID, bidder, "Success")
```
Comment (ashishshinde-pubm) [inline modules/pubmatic/vastunwrap/unwrap_service.go:47]: Please use constants for sucess/timeout/failure status

File: modules/pubmatic/vastunwrap/unwrap_service.go [hunk 2]
```diff
@@ -12,33 +12,46 @@ import (
...
+		return
+	}
+	if respStatus == "2" {
+		m.MetricsEngine.RecordRequestStatus(accountID, bidder, "Timeout")
+		return
```
Comment (ashishshinde-pubm) [inline modules/pubmatic/vastunwrap/unwrap_service.go:52]: if we returns from here then we will not log any statement like timeout

File: modules/pubmatic/vastunwrap/stats/metrics.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	metrics.Registry = cfg.Registry
+
+	metrics.requests = newCounter(cfg, metrics.Registry,
+		"vastunwrap_status",
+		"Count of vast unwrap requests labeled by publisher ID, bidder and status.",
+		[]string{"pubID", "bidder", "status"})
+
+	metrics.requestTime = newHistogramVec(cfg, metrics.Registry,
+		"vastunwrap_request_time",
+		"Time taken to serve the request in seconds", []string{"pubID", "bidder"},
```
Comment (ashishshinde-pubm) [inline modules/pubmatic/vastunwrap/stats/metrics.go:38]: can we rephrase this description ?
Time taken to unwrap vast request in seconds/mille-seconds. 

Please check if you are recording the time in seconds or milli-second

File: modules/pubmatic/vastunwrap/stats/metrics.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	metrics.Registry = cfg.Registry
+
+	metrics.requests = newCounter(cfg, metrics.Registry,
+		"vastunwrap_status",
+		"Count of vast unwrap requests labeled by publisher ID, bidder and status.",
+		[]string{"pubID", "bidder", "status"})
+
+	metrics.requestTime = newHistogramVec(cfg, metrics.Registry,
+		"vastunwrap_request_time",
+		"Time taken to serve the vast unwrap request in Milliseconds", []string{"pubID", "bidder"},
```
Comment (pm-jaydeep-mohite) [inline modules/pubmatic/vastunwrap/stats/metrics.go:38]: Can we add more buckets like 600, 700, 800, 900 and 1000 and above ?

File: modules/pubmatic/vastunwrap/stats/mock/mock.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
+// Code generated by MockGen. DO NOT EDIT.
+// Source: github.com/PubMatic-OpenWrap/prebid-server/modules/pubmatic/vastunwrap/stats (interfaces: MetricsEngine)
+
+// Package mock_stats is a generated GoMock package.
+package mock
+
+import (
+	gomock "github.com/golang/mock/gomock"
+	reflect "reflect"
```
Comment (pm-jaydeep-mohite) [inline modules/pubmatic/vastunwrap/stats/mock/mock.go:9]: Do we need Alias for reflect and time package ?

File: modules/pubmatic/vastunwrap/stats/metrics.go (hunk 4 above)
Comment (Pubmatic-Supriya-Patil) [reply modules/pubmatic/vastunwrap/stats/metrics.go:38]: Actually we have timeout of 200 Milliseconds so I have added timeout upto 500. If required I will add all of above values for timeout bucket

File: modules/pubmatic/vastunwrap/stats/mock/mock.go (hunk 5 above)
Comment (Pubmatic-Supriya-Patil) [reply modules/pubmatic/vastunwrap/stats/mock/mock.go:9]: This is auto generated mock file using mockgen.

**Review by pm-jaydeep-mohite — APPROVED** [review]
LGTM

**Review by ashishshinde-pubm — APPROVED** [review]
LGTM

**Pull request merged**